serving/samples/secrets-go/README.md

```diff
@@ -0,0 +1,286 @@
+               #   below. This can be changed if both places are changed.
+               #  - `robot.json` is determined by the "key" that is used to hold the
+               #   secret content in the Kubernetes secret.  This can be changed
+               #   if both places are changed.
+               - name: GOOGLE_APPLICATION_DEFAULT
+                 value: /var/secret/robot.json
+
+               # This section specified where in the container we want the
+               # volume containing our secret to be mounted.
+               volumeMounts:
+               - name: robot-secret
+                 mountPath: /var/secret
+
+             # This section attaches the secret "google-robot-secret" to
+             # the Pod holding the user container.
+             volumes:
+             - name: robot-secret
+               secret:
+                 secretName: google-robot-secret
+   ```
+
+## Building and deploying the sample
+
+Once you have recreated the sample code files (or used the files in the sample
+folder) you're ready to build and deploy the sample app.
+
+1. Use Docker to build the sample code into a container. To build and push with
+   Docker Hub, run these commands replacing `{username}` with your Docker Hub
+   username:
+
+   ```shell
+   # Build the container on your local machine
+   docker build -t {username}/secrets-go .
+
+   # Push the container to docker registry
+   docker push {username}/secrets-go
+   ```
+
+1. After the build has completed and the container is pushed to docker hub, you
+   can deploy the app into your cluster. Ensure that the container image value
+   in `service.yaml` matches the container you built in the previous step. Apply
+   the configuration using `kubectl`:
+
+   ```shell
+   kubectl apply --filename service.yaml
+   ```
+
+1. Now that your service is created, Knative will perform the following steps:
+    - Create a new immutable revision for this version of the app.
+    - Network programming to create a route, ingress, service, and load balance
+    for your app.
+    - Automatically scale your pods up and down (including to zero active pods).
+
+1. Run the following command to find the external IP address for your service.
+   The ingress IP for your cluster is returned. If you just created your
+   cluster, you might need to wait and rerun the command until your service gets
+   asssigned an external IP address.
+
+   ```shell
+   # In Knative 0.2.x and prior versions, the `knative-ingressgateway` service was used instead of `istio-ingressgateway`.
+   INGRESSGATEWAY=knative-ingressgateway
+
+   # The use of `knative-ingressgateway` is deprecated in Knative v0.3.x.
+   # Use `istio-ingressgateway` instead, since `knative-ingressgateway`
+   # will be removed in Knative v0.4.
+   if kubectl get configmap config-istio -n knative-serving &> /dev/null; then
+       INGRESSGATEWAY=istio-ingressgateway
+   fi
+
+   kubectl get svc $INGRESSGATEWAY --namespace istio-system
```

Comment:
`kubectl get svc $INGRESSGATEWAY --namespace istio-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}'` ?
Or just
EXTERNAL_IP_ADDRESS=$(kubectl get svc $INGRESSGATEWAY --namespace istio-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}')

Reply:
This is consistent with the `helloworld-go` (and numerous other) docs we have, so I'm going to bias for consistency here.